### ISSUE TITLE
Moment.operations and Moment.qubits should be read-only properties

### ISSUE THREAD

**Issue by mpharrigan** [quantumlib/Cirq#2359]
You can do some pretty crazy things using public attributes. I recommend prefixing `operations` and `qubits` with underscores and exposing via a `@property`

```python
import cirq
q1, q2 = cirq.LineQubit.range(2)
circuit = cirq.Circuit(cirq.X(q1))
circuit2 = circuit.copy()
mom = circuit.moments[0]
mom.operations += (cirq.Y(q2),)
mom.qubits |= {q2}
print(circuit)
print('---------')
print(circuit2)
```
```
0: ───X───

1: ───Y───
---------
0: ───X───

1: ───Y───
```

**Comment by Strilanc** [comment]
Agreed.

**Comment by jitendrs** [comment]
do you want to restrict the attribute assignment with the property descriptor?

**Comment by Strilanc** [comment]
Yes

**Comment by jitendrs** [comment]
Please assign it to me @Strilanc

**Comment by jitendrs** [comment]
Thanks @kevinsung

**Comment by jitendrs** [comment]
@Strilanc do you want me to code setter function as well for qubits? 
After looking at code in moment.py. There is a problem with variable qubits. It has been holding assignment value in the constructor. 
It will throw attribute error if you don't write the setter method. other easiest solution would be creating new function for setting affected_qubits. Let me know what you think

**Comment by Strilanc** [comment]
Having a setter would defeat the purpose. In the constructor you should use the private attributes instead of the properties.

**Comment by jitendrs** [comment]
Please close this PR. as requested code changes merged into master from PR #2391

**Comment by jitendrs** [comment]
@kevinsung Thanks.
